system/modules/core/pages/PageRegular.php

```diff
@@ -87,6 +89,7 @@ public function generate($objPage, $blnCheckRequest=false)
 			{
 				while ($objModules->next())
```

Comment:
You are right and this probably applies to all places where we are traversing a model collection.